scripts/check_type_hints.py

```
@@ -194,13 +225,98 @@ def add_future_import(file_path: Path) -> bool:
             )
             return False
 
+        content = file_path.read_text(encoding="utf-8")
+        lines = content.splitlines(keepends=True)
+
+        insert_index = _find_insert_index(lines, content)
+        _insert_import_line(lines, insert_index)
+
         file_path.write_text("".join(lines), encoding="utf-8")
         return True
```

Ruff (0.14.7)

Consider moving this statement to an `else` block

(TRY300)